api/numaresourcesoperator/v1/numaresourcesoperator_types.go

```diff
@@ -125,6 +125,28 @@ type NodeGroup struct {
 	PoolName *string `json:"poolName,omitempty"`
 }
 
+// NodeGroupStatus reports the status of a NodeGroup once matches an actual set of nodes and it is correctly processed
+// by the system. In other words, is not possible to have a NodeGroupStatus which does not represent a valid NodeGroup
+// which in turn correctly references unambiguously a set of nodes in the cluster.
+// Hence, if a NodeGroupStatus is published, its `Name` must be present, because it refers back to a NodeGroup whose
+// config was correctly processed in the Spec. And its DaemonSet will be nonempty, because matches correctly a set
+// of nodes in the cluster. The Config is best-effort always represented, possibly reflecting the system defaults.
+// If the system cannot process a NodeGroup correctly from the Spec, it will report Degraded state in the top-level
+// condition, and will provide details using the aforementioned conditions.
+type NodeGroupStatus struct {
+	// DaemonSet of the configured RTEs, for this node group
+	//+operator-sdk:csv:customresourcedefinitions:type=status,displayName="RTE DaemonSets"
+	DaemonSet *NamespacedName `json:"daemonsets,omitempty"`
```

Comment:
not completely sure. If we publish a `NodeGroupStatus`, it MUST have a valid DaemonSet.

Reply:
I'm not sure I understand where this type can be improved. Can you explain more please?

Reply:
The type is fine, but this is likely (not 100% sure yet) mandatory, so can be a field, not a pointer

Reply:
correct, DS isn't optional, I'll make this a struct type instead. thanks!

Reply:
please hold on a sec, decision not yet final and I want to take some time to review the API to avoid more churn

Reply:
So, the guiding principle here is https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required .

If we publish a `NodeGroupStatus`, is because the operator is managing that `NodeGroup` with a RTE.
While NROP exists and manages `NUMAResourcesOperator` objects, is meant to manage RTEs through daemonset. This field is thus required, not optional, thus it should *not* be a pointer *nor* it should be `omitempty`.

Reply:
Adjusted in the latest updates and updated controller logic when publishing NodeGroupStatus to be after all DSs and MCP are in sync. Thanks!